subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/ResolutionServiceResolver.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.use.resolve.service.internal;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.ResolveException;
+import org.gradle.api.internal.artifacts.DependencyResolutionServices;
+import org.gradle.api.internal.artifacts.configurations.ConfigurationContainerInternal;
+import org.gradle.api.internal.artifacts.configurations.ConfigurationInternal;
+import org.gradle.api.internal.initialization.ClassLoaderScope;
+import org.gradle.api.internal.plugins.PluginInspector;
+import org.gradle.api.specs.Specs;
+import org.gradle.internal.Factory;
+import org.gradle.internal.classpath.ClassPath;
+import org.gradle.internal.classpath.DefaultClassPath;
+import org.gradle.internal.exceptions.Contextual;
+import org.gradle.plugin.use.PluginId;
+import org.gradle.plugin.use.resolve.internal.ClassPathPluginResolution;
+import org.gradle.plugin.use.resolve.internal.PluginResolution;
+
+import java.io.File;
+import java.util.Set;
+
+public class ResolutionServiceResolver {
```

Comment:
There are quite a few changes to the plugin portal resolver which seem unrelated to this change. Could you try to remove these from the PR?